app/src/basedockwidget.cpp

```diff
@@ -34,8 +35,54 @@ BaseDockWidget::BaseDockWidget(QWidget* pParent)
                       "border-width: 1px; }");
     }
 #endif
+
+    mTitleBarWidget = new TitleBarWidget();
+    mNoTitleBarWidget = new QWidget();
```

Comment:
Will this `mNoTitleBarWidget` be deleted anywhere?

Reply:
Thanks! It's good to ensure that there is an object that owns the title bar widget so it can always be freed properly.